### PR TITLE
Windows updates

### PR DESCRIPTION
**Pull request by pato-melowntech**
This is breaking change. For compatibility use:
`utility::scanDir(dir) -> utility::id2path(utility::scanDir(dir))`

### CODE REVIEW

**Comment by pato-melowntech** [comment]
Adds:
- force forward slashes in scanDir on Windows
- utility::rename with workaround for Windows

File: utility/path.cpp [hunk 1]
```diff
@@ -90,4 +91,21 @@ fs::path lexically_relative(const fs::path &path, const fs::path &base)
...
+{
+#ifdef _WIN32
+    if (oldPath != newPath)
+    {
+        // use copy/delete instead of problematic rename
+        auto copyOptions(fs::copy_options::recursive
+                         | fs::copy_options::overwrite_existing
+                         | fs::copy_options::copy_symlinks);
+        copy(oldPath, newPath, copyOptions);
+        remove_all(oldPath);
```
Comment (zap150) [inline utility/path.cpp:104]: Does this work with symlinks as intented?

Comment (pato-melowntech) [reply utility/path.cpp:104]: It should, as `fs::copy_options::copy_symlinks` is specified in copy options.

Comment (pato-melowntech) [reply utility/path.cpp:104]: and for `remove_all`: Symlinks are not followed (symlink is removed, not its target).

File: utility/filesystem.cpp [hunk 2]
```diff
@@ -169,11 +169,20 @@ std::map<std::string, fs::path> scanDir(const fs::path &root)
...
 
-    return map;
+    return results;
+}
+
+std::map<std::string, fs::path> id2path(const std::vector<fs::path> &localPaths) {
+    std::map<std::string, fs::path> result;
+    for (const auto & local : localPaths) {
+        const auto id((local.parent_path() / local.stem()).generic_string());
+        result[id] = local;
```
[the remark this answers is not in the document]
Comment (zap150) [reply utility/filesystem.cpp:183]: Let's stick with the original behaviour.

**Review by zap150 — APPROVED** [review]

**Pull request merged**